src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -274,12 +289,31 @@ public bool TryWriteBytes(Span<byte> destination, out int bytesWritten)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void WriteIPv6Bytes(Span<byte> destination)
         {
-            Debug.Assert(_numbers != null && _numbers.Length == NumberOfLabels);
-            int j = 0;
-            for (int i = 0; i < NumberOfLabels; i++)
+            ushort[]? numbers = _numbers;
+            Debug.Assert(numbers != null && numbers.Length == NumberOfLabels);
+
+            if (BitConverter.IsLittleEndian)
             {
-                destination[j++] = (byte)((_numbers[i] >> 8) & 0xFF);
-                destination[j++] = (byte)((_numbers[i]) & 0xFF);
+                if (Vector128.IsHardwareAccelerated)
+                {
+                    Vector128<byte> bytes = Vector128.Shuffle(
+                        Vector128.LoadUnsafe(ref MemoryMarshal.GetArrayDataReference(numbers)).AsByte(),
+                        Vector128.Create((byte)1, 0, 3, 2, 5, 4, 7, 6, 9, 8, 11, 10, 13, 12, 15, 14));
```

Comment:
The same vector is used above too. To avoid this duplication one could
```c#
private static Vector128<byte> IPv6ShuffleVector() => Vector128.Create((byte)1, 0, 3, 2, 5, 4, 7, 6, 9, 8, 11, 10, 13, 12, 15, 14));

// ...
{
    Vector128<byte> bytes = Vector128.Shuffle(..., IPv6ShuffleVector());
}
// ...
```
Codegen will be the same as that cheap method will be inlined, thus the constant is read from data section.

Reply:
This won't be better in this case.

Due to `GT_CNS_VEC` being introduced late, we need the `Vector128.Create` call to be manually inlined for .NET 7 to recognize it as a constant for `shuffle`.

This is something that should get fixed in .NET 8

Reply:
> recognize it as a constant for `shuffle`

Ah, thanks for reminding me about that case 👍🏻